src/bootstrap/src/core/build_steps/compile.rs

```diff
@@ -466,15 +456,45 @@ pub fn std_crates_for_run_make(run: &RunConfig<'_>) -> Vec<String> {
     }
 }
 
+/// Tries to find LLVM's `compiler-rt` source directory, for building `library/profiler_builtins`.
+///
+/// Normally it lives in the `src/llvm-project` submodule, but if we will be using a
+/// downloaded copy of CI LLVM, then we try to use the `compiler-rt` sources from
+/// there instead, which lets us avoid checking out the LLVM submodule.
+fn compiler_rt_for_profiler(builder: &Builder<'_>) -> PathBuf {
+    // Try to use `compiler-rt` sources from downloaded CI LLVM, if possible.
+    if builder.config.llvm_from_ci {
+        // CI LLVM might not have been downloaded yet, so try to download it now.
+        builder.config.maybe_download_ci_llvm();
```

Comment:
I was a bit worried about this running in the dry run mode, but it seems like LLVM runs `maybe_download_ci_llvm` in dry run too, so it should be fine.

Reply:
It looks like `maybe_download_ci_llvm` has a `!self.dry_run()` guard before it actually downloads anything, so the actual download should be skipped.

(This does mean that dry-run and normal invocations might disagree on what path will be used for `compiler-rt`.)